### PR TITLE
fix(curriculum): prevent hardcoding palidrome checker answers

### LINKED ISSUE #54063: campers hardcoding solutions for final projects
The JavaScript final projects have an issue, I don't know if how they are written, there are really specific user stories and no general one. Anyway, it's scarely common that campers try hardcoding the solutions. Should we try to write tests that discourage this?

Or, alternatively, more generic userstories that explain the thing better?
Like, we have
> When the #search-input element contains the value Red and the #search-button element is clicked, an alert should appear with the text "Pokémon not found"

But there isn't a more generic user story that is explicit on the requirement.
> When the #search-input element contains a value that is not a Pokémon name...

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Ref #54063

This generates one random phrase consisting of both numbers and lowercase letters. It'd be extremely difficult to hard-code a solution for this test. It'd be a lot less work just solving the problem rather than predicting whatever is generated.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-palindrome-checker-project/build-a-palindrome-checker.md [hunk 1]
```diff
@@ -228,6 +228,30 @@ checkBtn.click();
 assert.strictEqual(resultEl.innerText.trim().replace(/[.,?!]+$/g, '').toLowerCase(), 'five|\_/|four is not a palindrome');
 ```
 
+When the `#text-input` element contains a palindrome of random letters and it's reversal, the `#result` element
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-palindrome-checker-project/build-a-palindrome-checker.md:231]:
```suggestion
When the `#text-input` element contains a palindrome created from a random sequence of alphanumeric characters, the `#result` element 
```

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-palindrome-checker-project/build-a-palindrome-checker.md [hunk 2]
```diff
@@ -228,6 +228,30 @@ checkBtn.click();
 assert.strictEqual(resultEl.innerText.trim().replace(/[.,?!]+$/g, '').toLowerCase(), 'five|\_/|four is not a palindrome');
 ```
 
+When the `#text-input` element contains a palindrome created from a random sequence of alphanumeric characters, the `#result` element 
+should correctly identify it as a palindrome.
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-palindrome-checker-project/build-a-palindrome-checker.md:232]:
```suggestion
When the `#text-input` element contains an alphanumeric palindrome, the `#result` element should correctly identify it as a palindrome. 
```
Is it needed to mention how this palindrome was created?

**Comment by gikf** [comment]
I think one more test is required, to cover generated non-palindrome. Otherwise there's option to hardcode the specific cases and anything else consider as a palindrome.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-palindrome-checker-project/build-a-palindrome-checker.md (hunk 2 above)
Comment (a2937) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-palindrome-checker-project/build-a-palindrome-checker.md:232]: I guess not.

**Review by majestic-owl448 — APPROVED** [review]
I was going to say that the not palindrome could be a palindrome, but you are removing used letters, good 👍

**Review by Ksound22 — APPROVED** [review]
LGTM 🚀

**Pull request merged**